sigma/plugins/roles/addselfcolorrole.py

```diff
@@ -0,0 +1,52 @@
+import discord
+from sigma.core.permission import check_man_roles
+from sigma.core.rolecheck import matching_role
+
+async def addselfcolorrole(cmd, message, args):
+    # if no args, send help message
+    if not args:
+        await message.channel.send(cmd.help())
+        return
+
+    # checks permissions
+    if not check_man_roles(message.author, message.channel):
+        out_content = discord.Embed(type='rich', color=0xDB0000, title='⛔ Insufficient Permissions. Server Admin Only.')
+        await message.channel.send(None, embed=out_content)
+        return
+
+    # create color and name from user input
+    color = args[0].lstrip('#').upper()
+
+    # check if color is a proper format
+    try:
+        int(color, 16) #if it's not hexadecimal, it will raise ValueError
+        if len(color) != 6:
+            raise ValueError
+    except ValueError:
+        out_content = discord.Embed(type='rich', color=0xFF9900, title='⚠ Error')
+        out_content.add_field(name='Invalid Input', value='A valid hexadecimal color code has numbers from 0–9 and letters from A–F, and has six characters.')
+        await message.channel.send(None, embed=out_content)
+        return
+
+    # check if role exists
+    exists = matching_role(message.guild, color)
+    if exists:
+        out_content = discord.Embed(type='rich', color=0xFF9900, title='⚠ Error')
+        out_content.add_field(name='Role Exists', value=f'A role with the name **{color}** already exists.')
+        await message.channel.send(None, embed=out_content)
+        return
+
+    # create role
+    discord_colour = discord.Colour.from_rgb(*tuple(int(color[i:i+2], 16) for i in (0, 2, 4)))
```

Comment:
I believe `*(int(color[i:i+2], 16) for i in (0, 2, 4))` is shorter and works as well, can you check?

Reply:
You right

Reply:
also this should work too `discord_colour = discord.Color(value=int(color, 16))`

Reply:
actually even `value` is extra `discord_colour = discord.Color(value=int(color, 16))`

Reply:
You right

Reply:
You right